### PR TITLE
Add Releases API endpoint

### PR DESCRIPTION
**Pull request by migmartri**
This is the backend effort associated with https://github.com/helm/monocular/issues/193

The goal of this PR is to offer Helm releases lifecycle via the REST API which can be enabled on demand. 

- [x] Refactor API operations grouping them via labels (repositories, releases and charts)

```
swagger/restapi/operations/
├── charts
│   ├── get_all_charts.go
│   ├── get_all_charts_parameters.go
│   ├── get_all_charts_responses.go
...
├── healthz.go
├── healthz_parameters.go
├── healthz_responses.go
├── monocular_api.go
├── releases
│   ├── create_release.go
│   ├── create_release_parameters.go
│   ├── create_release_responses.go
│   ├── get_all_releases.go
│   ├── get_all_releases_parameters.go
│   └── get_all_releases_responses.go
└── repositories
    ├── get_all_repos.go
    ├── get_all_repos_parameters.go
    └── get_all_repos_responses.go
```
- [x] Add releases operations (see #193 for more info)
- [x] Install Tiller if it does not exist in the Cluster
- [x] Connects to tiller via the tiller-deploy service or by portForwarding
- [x] Add /releases API endpoint

Response example:

```json
{
  "data": [
    {
      "attributes": {
        "chartName": "mariadb",
        "chartVersion": "0.5.9",
        "name": "quieting-coral",
        "namespace": "miguel",
        "status": "DEPLOYED",
        "updated": "Tue Mar  7 16:47:47 2017"
      },
      "id": "quieting-coral",
      "type": "release"
    },
    {
      "attributes": {
        "chartName": "mariadb",
        "chartVersion": "0.5.9",
        "name": "sweet-vulture",
        "namespace": "default",
        "status": "DEPLOYED",
        "updated": "Tue Mar  7 09:53:59 2017"
      },
      "id": "sweet-vulture",
      "type": "release"
    },
]
```
- [x] Launch charts via POST /releases

Supported parameters example:

```json
{ 
    "chartId": "stable/mariadb", # Required
    "chartVersion": "0.5.9", # Required
    "namespace": "miguel",
    "releaseName: "my-release",
    "dryRun": false
}
```

- [x] Add release API endpoint to expose extra information like notes and resources.
- [x] Delete charts via DELETE /releases/[release-name]
- [x] Add configuration option in backend to enable this feature
- [x] Update Helm chart to support this feature flag
- [x] Document this feature in `doc/configuration`

### CODE REVIEW

File: src/api/data/helm/client/client.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+// NewHelmClient returns the Helm implementation of data.Client
+func NewHelmClient() data.Client {
+	conf, _ := config.GetConfig()
+	return &helmClient{
+		portForward: conf.TillerPortForward,
+	}
+}
+
+func (c *helmClient) ensureTillerInstalled(namespace string, client *internalclientset.Clientset) error {
```
Comment (migmartri) [inline src/api/data/helm/client/client.go]: Install Tiller if does not exist.

File: src/api/data/helm/client/client.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+	err = c.ensureTillerInstalled(tillerNamespace, client)
+	if err != nil {
+		return nil, err
+	}
+
+	var tillerHost = fmt.Sprintf("%s.%s:%d", tillerServiceName, tillerNamespace, tillerPort)
+
+	if c.portForward {
+		tunnel, err := portforwarder.New(tillerNamespace, client, config)
```
Comment (migmartri) [inline src/api/data/helm/client/client.go]: If the developer enabled portForwarding, a tunnel will be created in every request. Default is to use the new tiller-deploy service.

File: src/api/data/helm/releases/releases.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
+package releases
+
+import (
+	log "github.com/Sirupsen/logrus"
```
Comment (jackfrancis) [inline src/api/data/helm/releases/releases.go]: We might want to standardize to `logrus` everywhere; we this introduction I think it's safe to say we have a heterogeneous logging implementation.

Comment (migmartri) [inline src/api/data/helm/releases/releases.go]: Thanks! :)

**Pull request merged**